homeassistant/components/rainbird/__init__.py

```diff
@@ -1,42 +1,89 @@
 """Support for Rain Bird Irrigation system LNK WiFi Module."""
 import logging
 
+from pyrainbird import RainbirdController
 import voluptuous as vol
 
+from homeassistant.components import binary_sensor, sensor, switch
+from homeassistant.const import (
+    CONF_FRIENDLY_NAME,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_TRIGGER_TIME,
+)
+from homeassistant.helpers import discovery
 import homeassistant.helpers.config_validation as cv
-from homeassistant.const import CONF_HOST, CONF_PASSWORD
+
+CONF_ZONES = "zones"
+
+SUPPORTED_PLATFORMS = [switch.DOMAIN, sensor.DOMAIN, binary_sensor.DOMAIN]
 
 _LOGGER = logging.getLogger(__name__)
 
+RAINBIRD_CONTROLLER = "controller"
 DATA_RAINBIRD = "rainbird"
 DOMAIN = "rainbird"
 
-CONFIG_SCHEMA = vol.Schema(
+SENSOR_TYPE_RAINDELAY = "raindelay"
+SENSOR_TYPE_RAINSENSOR = "rainsensor"
+# sensor_type [ description, unit, icon ]
+SENSOR_TYPES = {
+    SENSOR_TYPE_RAINSENSOR: ["Rainsensor", None, "mdi:water"],
+    SENSOR_TYPE_RAINDELAY: ["Raindelay", None, "mdi:water-off"],
+}
+
+TRIGGER_TIME_SCHEMA = cv.positive_int
```

Comment:
Why did we change?

Reply:
That doesn't work. The complex type was not serializable and allowed several interpretations of time what we don't need here. If we use that construct, than we need to detect type, parse it and finally transfer it to int representing minutes of irrigation.
Which is what we want to set up: minutes of irrigation.
So there will be useless overhead to convert minutes from config through some unwanted type to minutes.
So I reverted to previous functional state.

Reply:
It's not useless. We want the time options to use the same validator. If we need an integer in the end, we e convert to an integer in the end of the validation.

Please change again and add the last integer validator

Reply:
here it is